src/Features/CSharp/Portable/Completion/KeywordRecommenders/StackAllocKeywordRecommender.cs

```diff
@@ -16,20 +16,48 @@ public StackAllocKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            // type t = |
-            var token = context.TargetToken;
-            if (token.IsUnsafeContext())
+            var node = context.TargetToken.Parent;
+
+            // In case of an empty string
+            if (node == null)
+            {
+                return false;
+            }
+
+            // After a cast
+            if (context.TargetToken.IsKind(SyntaxKind.CloseParenToken) &&
+                (node.IsKind(SyntaxKind.ParenthesizedExpression) || node.IsKind(SyntaxKind.CastExpression)))
+            {
+                node = node.Parent;
+            }
+
+            // Inside a conditional expression: value ? stackalloc : stackalloc
+            while (node.IsKind(SyntaxKind.ConditionalExpression) &&
+                (context.TargetToken.IsKind(SyntaxKind.CloseParenToken) || context.TargetToken.IsKind(SyntaxKind.QuestionToken) || context.TargetToken.IsKind(SyntaxKind.ColonToken)))
```

Comment:
This is not handled by the above condition because getting the node parent `node = node.Parent` doesn't move the `context.TargetToken`. It is still pointing to the cast node.

---
In reply to: [156214413](https://github.com/dotnet/roslyn/pull/23655#discussion_r156214413) [](ancestors = 156214413)